go/v1beta1/storage/filtering/filtering.go

```diff
@@ -0,0 +1,150 @@
+package filtering
+
+import (
+	"encoding/json"
+
+	"github.com/google/cel-go/common"
+	"github.com/google/cel-go/parser"
+	expr "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+)
+
+// Query holds a parent query that carries the entire search query
+type Query struct {
+	Query interface{} `json:"query"`
+}
+
+// Bool holds a general query that carries any number of
+// Must and Should operations
+type Bool struct {
+	Bool interface{} `json:"bool"`
+}
+
+// Must holds a must operator which each equates to an AND operation
+type Must struct {
+	Must []interface{} `json:"must"`
+}
+
+// Should holds a should operator which equates to an OR operation
+type Should struct {
+	Should []interface{} `json:"should"`
+}
+
+// Term holds a comparison for equating two strings
+type Term struct {
+	Term map[string]string `json:"term"`
+}
```

Comment:
I think I'd probably add these `type` definitions to a dedicated `types.go` file.